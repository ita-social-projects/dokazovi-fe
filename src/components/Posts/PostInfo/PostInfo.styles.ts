import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(
  () => ({
    root: {
      display: 'flex',
    },
    topics: {
      listStyle: 'none',
      display: 'flex',
      marginRight: '20px',
      height: '32px',
      '& > *:not(:last-child)': {
        marginRight: '4px',
      },
      '& > li': {
        padding: '5px 12px',
        color: '#ffffff',
        backgroundColor: '#73DDFF',
      },
    },

    origins: {
      listStyle: 'none',
      display: 'flex',
      marginRight: '20px',
      height: '32px',

      '& > *:not(:last-child)': {
        marginRight: '4px',
      },
      '& > li': {
        padding: '5px 12px',
      },
      '& > *:first-child': {
        borderRightColor: '#FF5C00',
        borderRightStyle: 'solid',
        borderRightWidth: '1px',
      },
    },

    origin: {
      color: '#FF5C00',
    },

    createdAt: {
      color: '#767676',
      borderLeftColor: '#767676',
      borderLeftStyle: 'solid',
      borderLeftWidth: '1px',
    },

    icon: {
      '& > svg': {
        fill: '#767676',
        marginRight: '-20px',
      },
    },

    counter: {
      color: '#767676',
    },
  }),
  { name: 'PostInfo' },
);
