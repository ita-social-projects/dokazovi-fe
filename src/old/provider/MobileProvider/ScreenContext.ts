import { createContext } from 'react';

interface IScreenContext {
  mobile: boolean | null;
  tablet: boolean | null;
}

export const ScreenContext = createContext<IScreenContext>({
  mobile: null,
  tablet: null,
});
