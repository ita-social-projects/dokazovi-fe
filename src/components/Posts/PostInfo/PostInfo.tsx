import React from 'react';
import VisibilityIcon from '@material-ui/icons/Visibility';
import { IDirection, IOrigin, IPostType } from '../../../old/lib/types';
import { useStyles } from './PostInfo.styles';

export interface IPostInfo {
  info: {
    directions: IDirection[];
    origins: IOrigin[];
    type: IPostType;
    publishedAt: string;
    counter: number;
  };
}

export default function PostInfo({ info }: IPostInfo): JSX.Element {
  const { directions, origins, type, publishedAt, counter = 0 } = info;
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <ul className={classes.topics}>
        {directions && directions.map((el) => <li key={el.id}>{el.label}</li>)}
      </ul>
      <ul className={classes.origins}>
        {origins &&
          origins.map(
            (el) =>
              el.name && (
                <li className={classes.origin} key={el.id}>
                  {el.name}
                </li>
              ),
          )}
        {type && <li className={classes.origin}>{type.name}</li>}
        <li className={classes.createdAt}>{publishedAt}</li>
        <li className={classes.icon}>
          <VisibilityIcon fontSize="small" />
        </li>
        <li className={classes.counter}>{counter}</li>
      </ul>
    </div>
  );
}
