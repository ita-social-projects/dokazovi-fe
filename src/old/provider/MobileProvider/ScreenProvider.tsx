import React from 'react';
import { useMediaQuery, useTheme } from '@material-ui/core';
import { ScreenContext } from './ScreenContext';

export const ScreenProvider: React.FC = ({ children }) => {
  const theme = useTheme();
  const mobile = !useMediaQuery(theme.breakpoints.up('sm'));
  const tablet = !useMediaQuery(theme.breakpoints.up('md'));
return(
  <ScreenContext.Provider value={{ mobile,tablet }}>{children}</ScreenContext.Provider>
);
};
