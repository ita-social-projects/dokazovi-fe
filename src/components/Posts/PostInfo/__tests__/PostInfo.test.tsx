import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { ScreenContext } from 'old/provider/MobileProvider/ScreenContext';
import PostInfo, { IPostInfo } from '../PostInfo';

describe('PostInfo tests', () => {
  let mocks: IPostInfo;

  const renderComponentWithRouter = () => {
    const history = createMemoryHistory();

    render(
      <Router history={history}>
        <PostInfo info={mocks.info} />
      </Router>,
    );

    return { history };
  };

  beforeEach(() => {
    mocks = {
      info: {
        directions: [
          {
            id: 7,
            name: 'pediatrics',
            label: 'Педіатрія',
            color: '#993333',
            hasDoctors: true,
            hasPosts: true,
          },
          {
            id: 4,
            name: 'therapy',
            label: 'Терапія',
            color: '#ffee58',
            hasDoctors: true,
            hasPosts: true,
          },
        ],
        origins: [
          {
            id: 2,
            name: 'Медитека',
            parameter: null,
          },
        ],
        type: {
          id: 1,
          name: 'Стаття',
        },
        publishedAt: '05.10.2021',
        displayViews: 9,
      },
    };
  });

  it('should PostInfo render correctly', () => {
    const { asFragment } = render(<PostInfo info={mocks.info} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should main blocks be in the component', () => {
    render(<PostInfo info={mocks.info} />);

    expect(screen.getByTestId('topics')).toBeInTheDocument();
    expect(screen.getByTestId('origins')).toBeInTheDocument();
  });

  it('should onClick handler redirect to directions', () => {
    const { history } = renderComponentWithRouter();

    const items = screen.getAllByTestId('direction');
    expect(items[0]).toBeInTheDocument();

    userEvent.click(items[0]);
    expect(history.length).toBe(2);
    expect(history.location.pathname).toBe('/materials');
    expect(history.location.search).toBe('?directions=7');
  });

  it('should onClick handler redirect to origins', () => {
    const { history } = renderComponentWithRouter();

    const items = screen.getAllByTestId('origin');
    expect(items[0]).toBeInTheDocument();

    userEvent.click(items[0]);
    expect(history.length).toBe(2);
    expect(history.location.pathname).toBe('/materials');
    expect(history.location.search).toBe('?origins=2');
  });

  it('should onClick handler redirect to types', () => {
    const { history } = renderComponentWithRouter();

    const items = screen.getAllByTestId('type');
    expect(items[0]).toBeInTheDocument();

    userEvent.click(items[0]);
    expect(history.length).toBe(2);
    expect(history.location.pathname).toBe('/materials');
    expect(history.location.search).toBe('?types=1');
  });

  it('should mobile be with Views Counter', () => {
    const { asFragment } = render(
      <ScreenContext.Provider value={{ mobile: true, tablet: null }}>
        <PostInfo info={mocks.info} />
      </ScreenContext.Provider>,
    );

    expect(screen.queryByTestId('icon')).toBeInTheDocument();
    expect(screen.queryByTestId('counter')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should Skeleton be visible', () => {
    delete mocks.info.displayViews;

    const { asFragment } = render(<PostInfo info={mocks.info} />);

    expect(screen.getByTestId('skeleton')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
