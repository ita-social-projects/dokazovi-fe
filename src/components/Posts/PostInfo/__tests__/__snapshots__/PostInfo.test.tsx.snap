// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostInfo tests should PostInfo render correctly 1`] = `
<DocumentFragment>
  <div
    class="PostInfo-root-1"
  >
    <ul
      class="PostInfo-topics-2"
      data-testid="topics"
    >
      <li
        data-testid="direction"
      >
        Педіатрія
      </li>
      <li
        data-testid="direction"
      >
        Терапія
      </li>
    </ul>
    <ul
      class="PostInfo-origins-3"
      data-testid="origins"
    >
      <li
        class="PostInfo-origin-4"
        data-testid="origin"
      >
        Медитека
      </li>
      <li
        class="PostInfo-origin-4"
        data-testid="type"
      >
        Стаття
      </li>
      <li
        class="PostInfo-createdAt-5"
      >
        05.10.2021
      </li>
      <li
        class="PostInfo-icon-6"
        data-testid="icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
      </li>
      <li
        class="PostInfo-counter-7"
        data-testid="counter"
      >
        9
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`PostInfo tests should Skeleton be visible 1`] = `
<DocumentFragment>
  <div
    class="PostInfo-root-43"
  >
    <ul
      class="PostInfo-topics-44"
      data-testid="topics"
    >
      <li
        data-testid="direction"
      >
        Педіатрія
      </li>
      <li
        data-testid="direction"
      >
        Терапія
      </li>
    </ul>
    <ul
      class="PostInfo-origins-45"
      data-testid="origins"
    >
      <li
        class="PostInfo-origin-46"
        data-testid="origin"
      >
        Медитека
      </li>
      <li
        class="PostInfo-origin-46"
        data-testid="type"
      >
        Стаття
      </li>
      <li
        class="PostInfo-createdAt-47"
      >
        05.10.2021
      </li>
      <li
        class="PostInfo-icon-48"
        data-testid="icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
      </li>
      <li
        class="PostInfo-counter-49"
        data-testid="counter"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
          data-testid="skeleton"
          style="width: 40px; height: 20px;"
        />
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`PostInfo tests should mobile be without Views Counter 1`] = `
<DocumentFragment>
  <div
    class="PostInfo-root-36"
  >
    <ul
      class="PostInfo-topics-37"
      data-testid="topics"
    >
      <li
        data-testid="direction"
      >
        Педіатрія
      </li>
      <li
        data-testid="direction"
      >
        Терапія
      </li>
    </ul>
    <ul
      class="PostInfo-origins-38"
      data-testid="origins"
    >
      <li
        class="PostInfo-origin-39"
        data-testid="origin"
      >
        Медитека
      </li>
      <li
        class="PostInfo-origin-39"
        data-testid="type"
      >
        Стаття
      </li>
      <li
        class="PostInfo-createdAt-40"
      >
        05.10.2021
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
